File: web-simulator/js/apps/breathing.js
/** TinyOLED Desktop — Breathing Exercise */
const PHASES=[['NEFES AL',4],['TUT',7],['NEFES VER',8]];

export class BreathingApp {
  static NAME='breath'; static LABEL='Nefes'; static ICON='breath';
  constructor(onExit) {
    this.onExit=onExit; this.running=false; this.phase=0; this.start=0; this.cycles=0;
  }
  onUp(){} onDown(){}
  onSel() {
    if(!this.running){this.running=true;this.phase=0;this.start=performance.now()/1000;this.cycles=0;}
    else this.running=false;
  }
  onLong(){this.onExit();}
  update() {
    if(!this.running) return;
    const elapsed=performance.now()/1000-this.start;
    if(elapsed>=PHASES[this.phase][1]) {
      this.phase++;
      if(this.phase>=PHASES.length){this.phase=0;this.cycles++;}
      this.start=performance.now()/1000;
    }
  }
  draw(fb) {
    if(!this.running) {
      fb.textCentered('Nefes Egzersizi',14);
      fb.textCentered('4-7-8 Teknigi',26);
      fb.textCentered('[SEL] Basla',42);
      return;
    }
    const[name,dur]=PHASES[this.phase];
    const elapsed=performance.now()/1000-this.start;
    const prog=Math.min(1,elapsed/dur), rem=Math.max(0,dur-elapsed);
    fb.textCentered(name,10);
    const cx=64,cy=38,mn=4,mx=22;
    let r;
    if(this.phase===0) r=Math.round(mn+(mx-mn)*prog);
    else if(this.phase===1) r=mx+Math.round(Math.sin(elapsed*6)*1.5);
    else r=Math.round(mx-(mx-mn)*prog);
    fb.circle(cx,cy,r,true,true);
    fb.circle(cx,cy,mx+2);
    fb.text(`${Math.floor(rem)}s`,110,34);
    fb.text(`Dongu:${this.cycles}`,2,56);
  }
}

File: apps/breathing.py
"""TinyOLED Desktop — Nefes Egzersizi Rehberi"""
import math, time
from core.framebuffer import Framebuffer
PHASES=[("NEFES AL",4.0),("TUT",7.0),("NEFES VER",8.0)]

class BreathingApp:
    NAME="breath"; LABEL="Nefes"; ICON="breath"
    def __init__(self, on_exit):
        self.on_exit=on_exit; self.running=False; self.phase=0; self.start=0; self.cycles=0
    def on_up(self): pass
    def on_down(self): pass
    def on_sel(self):
        if not self.running: self.running=True; self.phase=0; self.start=time.monotonic(); self.cycles=0
        else: self.running=False
    def on_long(self): self.on_exit()
    def update(self):
        if not self.running: return
        elapsed=time.monotonic()-self.start
        if elapsed>=PHASES[self.phase][1]:
            self.phase+=1
            if self.phase>=len(PHASES): self.phase=0; self.cycles+=1
            self.start=time.monotonic()
    def draw(self, fb: Framebuffer):
        if not self.running:
            fb.text_centered("Nefes Egzersizi",14); fb.text_centered("4-7-8 Teknigi",26)
            fb.text_centered("[SEL] Basla",42); return
        name,dur=PHASES[self.phase]; elapsed=time.monotonic()-self.start
        prog=min(1,elapsed/dur); rem=max(0,dur-elapsed)
        fb.text_centered(name,10); cx,cy=64,38; mn,mx=4,22
        if self.phase==0: r=int(mn+(mx-mn)*prog)
        elif self.phase==1: r=mx+int(math.sin(elapsed*6)*1.5)
        else: r=int(mx-(mx-mn)*prog)
        fb.circle(cx,cy,r,fill=True); fb.circle(cx,cy,mx+2)
        fb.text(f"{rem:.0f}s",110,34); fb.text(f"Dongu:{self.cycles}",2,56)
